tests/misc/test_vtpm_basic_operations.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import pytest
+import lib.commands as commands
+
+vtpm_signing_test_script = """#!/bin/env bash
+
+set -ex
+
+apt-get install -y tpm2-tools > /dev/null
+
+tpm2_selftest --fulltest
+
+tpm2_getrandom 32 > /dev/null
+
+TMPDIR=`mktemp -d`
+
+# Create an Endorsement primary key
+tpm2_createprimary --hierarchy e --key-context ${TMPDIR}/primary.ctx > /dev/null
+
+# Create key objects
+tpm2_create --key-algorithm rsa --public ${TMPDIR}/rsa.pub --private ${TMPDIR}/rsa.priv --parent-context \
+            ${TMPDIR}/primary.ctx > /dev/null
+
+# Load keys into the TPM
+tpm2_load --parent-context ${TMPDIR}/primary.ctx --public ${TMPDIR}/rsa.pub --private ${TMPDIR}/rsa.priv \
+          --key-context ${TMPDIR}/rsa.ctx > /dev/null
+
+# Delete loaded key files
+rm -f ${TMPDIR}/rsa.pub ${TMPDIR}/rsa.priv
+
+# Message to sign
+echo 'XCP-ng Rulez' > ${TMPDIR}/message.dat
+
+# Sign the message
+tpm2_sign --key-context ${TMPDIR}/rsa.ctx --hash-algorithm sha256 --signature ${TMPDIR}/sig.rssa \
+          ${TMPDIR}/message.dat > /dev/null
+
+# Verify signature
+tpm2_verifysignature --key-context ${TMPDIR}/rsa.ctx --hash-algorithm sha256 --message ${TMPDIR}/message.dat \
+                     --signature ${TMPDIR}/sig.rssa > /dev/null
+
+# Verify with another message
+echo "XCP-ng Still Rulez" > ${TMPDIR}/message.dat
+
+# Verify signature !!!!! THIS MUST FAIL !!!!!
+if tpm2_verifysignature --key-context ${TMPDIR}/rsa.ctx --hash-algorithm sha256 --message ${TMPDIR}/message.dat \
+                        --signature ${TMPDIR}/sig.rssa > /dev/null 2>&1; then
+    echo "Should not succeed"
+    exit 1
+fi
+
+rm -rf ${TMPDIR}
+"""
+
+def _vtpm_get_for_vm(host, vm_uuid):
+    try:
+        res = host.xe('vtpm-list', {'vm': vm_uuid})
+        return res.splitlines()[0].split(':')[1].strip()
+    except IndexError:
+        return None
+
+def _vtpm_create(host, vm_uuid):
+    logging.info("Creating vtpm for vm %s" % vm_uuid)
+    return host.xe('vtpm-create', {'vm-uuid': vm_uuid})
+
+def _vtpm_destroy(host, vtpm_uuid):
+    if vtpm_uuid is None:
+        return
+
+    logging.info("Destroying vtpm %s" % vtpm_uuid)
+    host.xe('vtpm-destroy', {'uuid': vtpm_uuid}, force=True)
+
+def _vtpm_init(host, uefi_vm):
+    vm = uefi_vm
+
+    # vm must be shutdown to attach a vtpm
+    if not vm.is_halted():
+        force = vm.is_paused() or vm.is_suspended()
+        vm.shutdown(verify=True, force=force)
+
+    vtpm_uuid = _vtpm_get_for_vm(host, vm.uuid)
+    # Fail if vtpm-destroy fails since it's not possible to attach more than
+    # one vtpm to a vm
+    _vtpm_destroy(host, vtpm_uuid)
+
+@pytest.mark.big_vm
+@pytest.mark.usefixtures("host_at_least_8_3")
+def test_vtpm(host, uefi_vm):
```

Comment:
I think this tests does too much for a single test. 

I'd separate this in two tests:
* test_create_and_destroy, where you just create a vTPM then destroy it. So that these operations are tested on their own.
* test_vtpm_in_vm which requires a new unix_vm_with_vtpm fixture (which in turn would require the existing unix_vm and uefi_vm fixtures, it's a cascade), the fixture handling the creation and destruction of the vTPM outside the test code itself.

Reply:
Done